From cd03a56e8ababb257cd0c6dea5e70d072a9d5ca7 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Mon, 27 May 2019 14:22:51 +0800
Subject: [PATCH] policy/modules/roles/sysadm: fix avc denials for sysadm_t

Fixes:
avc:  denied  { read write } for  pid=1453 comm="systemd"
path="socket:[21133]" dev="sockfs" ino=21133
scontext=root:sysadm_r:sysadm_t:s0
tcontext=system_u:system_r:init_t:s0-s15:c0.c1023
tclass=unix_stream_socket permissive=0

avc:  denied  { prog_load } for  pid=1453 comm="systemd"
scontext=root:sysadm_r:sysadm_t:s0 tcontext=root:sysadm_r:sysadm_t:s0
tclass=bpf permissive=0

avc:  denied  { setrlimit } for  pid=1424 comm="systemd"
scontext=root:sysadm_r:sysadm_t:s0 tcontext=root:sysadm_r:sysadm_t:s0
tclass=process permissive=1

avc:  denied  { use } for  pid=1428 comm="systemd" path="socket:[22663]"
dev="sockfs" ino=22663 scontext=root:sysadm_r:sysadm_t:s0
tcontext=system_u:system_r:init_t:s0-s15:c0.c1023 tclass=fd permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/roles/sysadm.te | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/policy/modules/roles/sysadm.te b/policy/modules/roles/sysadm.te
index ac5239d83..3ccfbd053 100644
--- a/policy/modules/roles/sysadm.te
+++ b/policy/modules/roles/sysadm.te
@@ -60,6 +60,13 @@ selinux_read_policy(sysadm_t)
 userdom_manage_user_home_dirs(sysadm_t)
 userdom_home_filetrans_user_home_dir(sysadm_t)
 
+allow sysadm_t self:bpf { map_create map_read map_write prog_load prog_run };
+allow sysadm_t self:capability audit_write;
+allow sysadm_t self:system reload;
+allow sysadm_t self:process setrlimit;
+init_rw_inherited_stream_socket(sysadm_t)
+init_use_fds(sysadm_t)
+
 ifdef(`direct_sysadm_daemon',`
 	optional_policy(`
 		init_run_daemon(sysadm_t, sysadm_r)
-- 
2.17.1

